lib/xclarity_client/xclarity_base.rb

```diff
@@ -84,5 +96,33 @@ def do_delete (uri="")
         req.headers['Content-Type'] = 'application/json'
       end
     end
+
+    def build_token_session(conf)
+      $lxca_log.info 'XClarityClient::XClarityBase build_session', 'Starting the session...'
+      @response = @conn.post do |request|
+        request.url SESSION_URI
+        request.headers['Content-Type'] = 'application/json'
+        request.body = { :UserId => conf.username,
+                         :password => conf.password }.to_json
+      end
+      raise Faraday::Error::ConnectionFailed unless @response.success?
+    rescue => error
+      translation_error(error)
+    end
+
+    def translation_error(error)
+      cause = error.cause.to_s.downcase
+      if cause.include?('connection refused')
+        raise XClarityClient::Error::ConnectionRefused
+      elsif cause.include?('name or service not known')
+        raise XClarityClient::Error::HostnameUnknown
+      elsif @response.nil?
```

Comment:
For a more [pure](https://en.wikipedia.org/wiki/Pure_function) approach, I think that will be a good idea to receive the  `@response` as a param